msgpack-core/src/main/java/org/msgpack/core/MessagePacker.java

```diff
@@ -191,12 +173,18 @@ public void close()
         }
     }
 
-    private void ensureCapacity(int numBytesToWrite)
+    private void ensureCapacity(int mimimumSize)
             throws IOException
     {
-        if (buffer == null || position + numBytesToWrite >= buffer.size()) {
-            flush();
-            buffer = out.next(Math.max(config.packerBufferSize, numBytesToWrite));
+        if (buffer == null) {
+            buffer = out.next(mimimumSize);
+        }
+        else if (position + mimimumSize >= buffer.size()) {
+            out.writeBuffer(position);
```

Comment:
These codes are almost same as `flush()`. How about moving the both codes into a new method?